File: r-ggplot2, python-matplotlib, javascript-d3, tableau, excel/viscars.R
library(tidyverse)
mtcars_sample <- read.csv("r-ggplot2, python-matplotlib, javascript-d3, tableau, excel/javascript-d3/cars-sample.csv", header=TRUE)
ggplot(mtcars_sample, aes(x=Weight, y=MPG)) +
  geom_point(aes(colour=Manufacturer, size=Weight), alpha=0.5)
ggsave("img/ggplot2.png")

File: r-ggplot2, python-matplotlib, javascript-d3, tableau, excel/viscars.py
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns


def main():
    plt.style.use('ggplot')
    mtcars = pd.read_csv("public/cars-sample.csv")
    cmap = {"bmw": "#f3b7b4",
            "ford": "#cdce8e",
            "honda": "#95d9bb",
            "mercedes": "#93d3f4",
            "toyota": "#eeb2f2"}
    _, ax = plt.subplots()
    for key, group in mtcars.groupby("Manufacturer"):
        group.plot.scatter(ax=ax, x="Weight", y="MPG", label=key, color=cmap[key], alpha=0.5)
    plt.xlabel("Weight")
    plt.xticks(range(1000, 5000, 1000))
    plt.ylabel("MPG")
    plt.yticks(range(10, 50, 10))
    plt.show()


if __name__ == '__main__':
    main()
